src/controller/python/ChipDeviceController-StorageDelegate.cpp

```diff
@@ -107,8 +108,8 @@ CHIP_ERROR StorageAdapter::SyncGetKeyValue(const char * key, void * value, uint1
 
 CHIP_ERROR StorageAdapter::SyncSetKeyValue(const char * key, const void * value, uint16_t size)
 {
+    ReturnErrorCodeIf(((value == nullptr) && (size != 0)), CHIP_ERROR_INVALID_ARGUMENT);
     ChipLogDetail(Controller, "StorageAdapter::SetKeyValue: Key = %s, Value = %p (%u)", key, value, size);
-    mStorage[key] = std::string(static_cast<const char *>(value), size);
```

Comment:
I guess with that we can also remove the `mStorage` field in the `StorageAdapter` class?